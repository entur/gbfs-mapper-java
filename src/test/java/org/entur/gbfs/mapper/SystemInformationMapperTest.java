package org.entur.gbfs.mapper;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.annotations.SnapshotName;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;

import java.io.IOException;
import java.net.URL;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;

@ExtendWith({SnapshotExtension.class})
class SystemInformationMapperTest {
    private static final ObjectMapper objectMapper = new ObjectMapper();

    private Expect expect;

    @SnapshotName("gbfs_v2_3_to_v3_0_system_information_file_snapshot")
    @Test
    void testMapSystemInformationFile() throws IOException {
        URL resource = getClass().getClassLoader().getResource("fixtures/v2_3/system_information.json");
        org.entur.gbfs.v2_3.system_information.GBFSSystemInformation testSubject = objectMapper.readValue(resource, org.entur.gbfs.v2_3.system_information.GBFSSystemInformation.class);
        org.entur.gbfs.v3_0.system_information.GBFSSystemInformation mapped = GBFSMapper.INSTANCE.map(testSubject, "en");
        assertDoesNotThrow(() -> {
            expect
                    .serializer("json")
                    .toMatchSnapshot(mapped);
        });
    }

    @SnapshotName("gbfs_v3_0_to_v2_3_system_information_file_snapshot")
    @Test
    void testMapSystemInformationFileInverse() throws IOException {
        URL resource = getClass().getClassLoader().getResource("fixtures/v3_0/system_information.json");
        org.entur.gbfs.v3_0.system_information.GBFSSystemInformation testSubject = objectMapper.readValue(resource, org.entur.gbfs.v3_0.system_information.GBFSSystemInformation.class);
        org.entur.gbfs.v2_3.system_information.GBFSSystemInformation mapped = GBFSMapper.INSTANCE.map(testSubject, "en");
        assertDoesNotThrow(() -> {
            expect
                    .serializer("json")
                    .toMatchSnapshot(mapped);
        });
    }
}
