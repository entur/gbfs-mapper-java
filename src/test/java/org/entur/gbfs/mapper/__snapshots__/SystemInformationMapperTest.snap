gbfs_v2_3_to_v3_0_system_information_file_snapshot=[
  {
    "last_updated": "2021-12-30T17:59:23.000+00:00",
    "ttl": 1800,
    "version": "3.0-RC2",
    "data": {
      "system_id": "example_cityname",
      "languages": [
        "en"
      ],
      "name": [
        {
          "text": "Example Bike Rental",
          "language": "en"
        }
      ],
      "short_name": [
        {
          "text": "Example Bike",
          "language": "en"
        }
      ],
      "operator": [
        {
          "text": "Example Sharing, Inc",
          "language": "en"
        }
      ],
      "url": "https://www.example.com",
      "purchase_url": "https://www.example.com",
      "start_date": "2010-06-10",
      "phone_number": "+18005551234",
      "email": "customerservice@example.com",
      "feed_contact_email": "datafeed@example.com",
      "timezone": "America/Chicago",
      "license_url": "https://www.example.com/data-license.html",
      "brand_assets": {
        "brand_last_modified": "2021-06-15",
        "brand_terms_url": "https://www.example.com/assets/brand.pdf",
        "brand_image_url": "https://www.example.com/assets/brand_image.svg",
        "brand_image_url_dark": "https://www.example.com/assets/brand_image_dark.svg",
        "color": "#C2D32C"
      },
      "terms_url": [
        {
          "text": "https://www.example.com/en/terms",
          "language": "en"
        }
      ],
      "terms_last_updated": "2021-06-21",
      "privacy_url": [
        {
          "text": "https://www.example.com/en/privacy-policy",
          "language": "en"
        }
      ],
      "privacy_last_updated": "2019-01-13",
      "rental_apps": {
        "android": {
          "store_uri": "https://play.google.com/store/apps/details?id=com.example.android",
          "discovery_uri": "com.example.android://"
        },
        "ios": {
          "store_uri": "https://apps.apple.com/app/apple-store/id123456789",
          "discovery_uri": "com.example.ios://"
        }
      }
    }
  }
]


gbfs_v3_0_to_v2_3_system_information_file_snapshot=[
  {
    "last_updated": 1640887163,
    "ttl": 1800,
    "version": "2.3",
    "data": {
      "system_id": "example_cityname",
      "language": "en",
      "name": "Example Bike Rental",
      "short_name": "Example Bike",
      "operator": "Example Sharing, Inc",
      "url": "https://www.example.com",
      "purchase_url": "https://www.example.com",
      "start_date": "2010-06-10",
      "phone_number": "+18005551234",
      "email": "customerservice@example.com",
      "feed_contact_email": "datafeed@example.com",
      "timezone": "America/Chicago",
      "license_url": "https://www.example.com/data-license.html",
      "brand_assets": {
        "brand_last_modified": "2021-06-15",
        "brand_terms_url": "https://www.example.com/assets/brand.pdf",
        "brand_image_url": "https://www.example.com/assets/brand_image.svg",
        "brand_image_url_dark": "https://www.example.com/assets/brand_image_dark.svg",
        "color": "#C2D32C"
      },
      "terms_url": "https://www.example.com/en/terms",
      "terms_last_updated": "2021-06-21",
      "privacy_url": "https://www.example.com/en/privacy-policy",
      "privacy_last_updated": "2019-01-13",
      "rental_apps": {
        "android": {
          "store_uri": "https://play.google.com/store/apps/details?id=com.example.android",
          "discovery_uri": "com.example.android://"
        },
        "ios": {
          "store_uri": "https://apps.apple.com/app/apple-store/id123456789",
          "discovery_uri": "com.example.ios://"
        }
      }
    }
  }
]